tests/test_declarative_assembly_format.py

```diff
@@ -245,6 +246,49 @@ class PropOp(IRDLOperation):
     check_equivalence(program, generic_program, ctx)
 
 
+################################################################################
+# Attribute variables                                                          #
+################################################################################
+
+
+@irdl_op_definition
+class OpWithAttr(IRDLOperation):
+    name = "test.one_attr"
+
+    attr = attr_def(Attribute)
+    assembly_format = "$attr attr-dict"
+
+
+@pytest.mark.parametrize(
+    "program, generic_program",
+    [
+        ("test.one_attr i32", '"test.one_attr"() {"attr" = i32} : () -> ()'),
+        (
+            'test.one_attr i32 {"attr2" = i64}',
+            '"test.one_attr"() {"attr" = i32, "attr2" = i64} : () -> ()',
+        ),
+    ],
+)
+def test_standard_attr_directive(program: str, generic_program: str):
+    ctx = MLContext()
+    ctx.load_op(OpWithAttr)
+
+    check_equivalence(program, generic_program, ctx)
+    check_roundtrip(program, ctx)
+
+
+def test_attr_variable_shadowed():
+    ctx = MLContext()
+    ctx.load_op(OpWithAttr)
+
+    parser = Parser(ctx, "test.one_attr i32 {attr = i64}")
+    with pytest.raises(
+        ParseError,
+        match="attributes attr are defined in other parts",
+    ):
+        parser.parse_operation()
```

Comment:
This is where MLIR silently does something instead of erroring out, right?

Reply:
Yep, MLIR will override one of them!

Reply:
Alright! Let's hope we can peacefully enforce that then ahah